Cabal/doc/getting-started.rst

```diff
@@ -0,0 +1,159 @@
+Getting Started with Haskell and Cabal
+======================================
+
+Installing the Haskell toolchain
+--------------------------------
+
+To install the Haskell toolchain follow the `ghcup instructions
+<https://www.haskell.org/ghcup/>`__.
+
+
+Creating a new application
+--------------------------
+
+Let's start by creating a simple Haskell application from scratch where we'll
+learn about a Haskell package's directory structure, how to run the executable,
+and how to add external dependencies.
+
+
+Initializing the application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start by creating a ``myfirstapp`` directory to hold the project files:
+
+.. code-block:: console
+
+    $ mkdir myfirstapp && cd myfirstapp
+
+Once you have an empty directory we can initialize our package:
+
+.. code-block:: console
+
+    $ cabal init
+
+This will generate the following files:
+
+.. code-block:: console
+
+    $ ls
+    CHANGELOG.md
+    Main.hs
+    myfirstapp.cabal
+
+
+``Main.hs`` is where your package's code lives. By default ``cabal init``
+creates an executable with the same name as the package ``myfirstapp`` in this
+case, you can instruct ``cabal init`` to generate just a library (with
+``--lib``) or both a library and executable with (``--libandexe``); for the full
+set of options see ``cabal init --help``.
+
+``myfirstapp.cabal`` is Cabal's metadata file which describes your package and
+its dependencies. We'll be updating this file in a little bit when we add an
+external dependency to our package.
+
+.. note:: By default the name of the current directory will be used for the
+	  package name, you can change this with ``-p <package-name>``.
+
+
+Running the application
+^^^^^^^^^^^^^^^^^^^^^^^
+
+As mentioned above, ``cabal init`` with no arguments generates a package with a
+single executable that prints ``"Hello, Haskell!"`` to the terminal. To run the
+executable enter the following command:
+
+``cabal run :myfirstapp``
+
+You should see the following output in the terminal:
+
+.. code-block:: console
+
+     $ cabal run :myfirstapp
+     ...
+     Hello, Haskell!
+
+.. note:: The ``:`` prefix in ``:myfirstapp`` signifies that the the
+	  ``myfirstapp`` target is part of the current package.
+
+Notice that we didn't need to run a `build` command before ``cabal run``, this
+is because ``cabal run`` first determines if the code needs to be re-built
+before running the executable. If you just want to build a target you can do so
+with ``cabal build``:
+
+``cabal build :myfirstapp``
+
+
+Adding dependencies
+^^^^^^^^^^^^^^^^^^^
+
+Next we'll add an external dependency to our application. The unit of
+shareability in the Haskell ecosystem is a `package`, the database of available
+Haskell packages is called `Hackage <https://hackage.haskell.org/>`__.
+
+In our application, we'll use a package called `haskell-say
+<https://hackage.haskell.org/package/haskell-say>`__ to print text to the
```

Comment:
I wonder if there are any "better" example package. If users go to see that package they'll find `package.yaml` and `stack.yaml` and no CI setup and ...

If we put something at the very front (*the first 3rd party package a new user may see*), we have to put something of the highest quality. I'm biased, so I won't suggest any specific package here.

Reply:
I was looking for something very small, with zero dependencies, that did something fun, that works entirely in the terminal. We could easily add a test or two and a CI setup to that package if we want it to showcase everything (I can send a PR this week).

Reply:
And I think as far as `package.yaml` and `stack.yaml` go, these are both very widely used tools in the community today (regardless of personal opinions of them), so exposing a newcomer to them isn't necessarily a bad idea.